client/src/main/java/com/paypal/selion/internal/platform/grid/MobileTestSession.java

```diff
@@ -212,14 +224,21 @@ public void initializeTestSession(InvokedMethodInformation method) {
         }
 
         boolean appPathProvided = StringUtils.isNotBlank(appPath);
+        
+        //App run and browser run should be mutually exclusive. If appName or appPath is provided disregard
+        //the browser parameter
+        
+        if(StringUtils.isNotBlank(appName) || StringUtils.isNotBlank(appPath)){
+        	this.browserName = ""; //to nullify the default value *firefox
+        }
 
         checkArgument(!(mobileNodeType != MobileNodeType.APPIUM && appPathProvided),
                 "appPath can be specified for appium only, Please specify appName instead of appPath");
 
         checkArgument(
-                StringUtils.isNotBlank(appName) ^ StringUtils.isNotBlank(appPath),
-                "Either you have provided both appPath and appName or you have specified nothing. Please specify either "
-                + "appPath or appName");
+                StringUtils.isNotBlank(appName) ^ StringUtils.isNotBlank(appPath) ^ StringUtils.isNotBlank(browserName),
+                "Either you have provided appPath, appName and browserName all three of them or you have specified nothing. Please specify either "
```

Comment:
I suggest the text change to the following...
`Either you have provided appPath, appName, and browserName all at once or you have specified nothing. Please specify one of appPath, appName, or browserName`